llama-index-core/llama_index/core/instrumentation/dispatcher.py

```diff
@@ -216,57 +170,14 @@ def span_exit(
             else:
                 c = c.parent
 
-    def get_dispatch_event(self) -> EventDispatcher:
-        """Get dispatch_event for firing events within the context of a span.
-
-        This method should be used with @dispatcher.span decorated
-        functions only. Otherwise, the span_id should not be trusted, as the
-        span decorator sets the span_id.
-        """
-        with self.lock:
-            span_id = self.current_span_id
-        dispatch_event: EventDispatcher = partial(self.event, span_id=span_id)
-        return dispatch_event
-
-    def _get_parent_update_span_ctx_var(self, id_: str, current_task_name: str):
-        """Helper method to get parent id from the appropriate async contextvar."""
-        current_thread = threading.get_ident()
-        thread_span_ctx = span_ctx_var.get().copy()
-        span_ctx = thread_span_ctx[current_thread]
-        if current_task_name not in span_ctx:
-            parent_id = None
-            span_ctx[current_task_name] = [id_]
-        else:
-            parent_id = span_ctx[current_task_name][-1]
-            span_ctx[current_task_name].append(id_)
-        span_ctx_var.set(thread_span_ctx)
-
-        return parent_id
-
-    def _pop_span_from_ctx_var(self, current_task_name: str) -> None:
-        """Helper method to pop completed/dropped span from async contextvar."""
-        current_thread = threading.get_ident()
-        thread_span_ctx = span_ctx_var.get().copy()
-        span_ctx = thread_span_ctx[current_thread]
-
-        span_ctx[current_task_name].pop()
-        if len(span_ctx[current_task_name]) == 0:
-            del span_ctx[current_task_name]
-        span_ctx_var.set(thread_span_ctx)
-
     def span(self, func):
         @wrapt.decorator
         def wrapper(func, instance, args, kwargs):
             bound_args = inspect.signature(func).bind(*args, **kwargs)
             id_ = f"{func.__qualname__}-{uuid.uuid4()}"
-            with self.lock:
-                self.set_current_span_id(id_)
-            with self.root.lock:
-                self.root.set_current_span_id(id_)
-
-            # get parent_id (thread-safe)
-            parent_id = self._get_parent_update_span_ctx_var(id_, DEFAULT_SYNC_KEY)
 
+            token = active_span_id.set(id_)
+            parent_id = token.old_value
```

Comment:
@RogerHYang: I'm running into some trouble when trying to execute your code with the Basic Usage Notebook, particularly when it gets to Streaming section.

If `token.old_value` returns `Token.MISSING` marker object indicating that the context var had not been set before (https://docs.python.org/3/library/contextvars.html#contextvars.Token.old_value), but I don't believe that to be true.

Reply:
I'll look into it. I suspect it's because of the async nature of notebook. That's my oversight.

Reply:
No worries at all! Thanks very much.

Reply:
OK. The root cause of the issue is that `llama-index-agent-openai` package is not from this PR and so is still using the regular `Thread`. Therefore the `contextvar` value of `None` didn't get carried cross the thread barrier when the openai agent runs. I have updated this PR to handle `Token.MISSING` so that it would be backward-compatible.

Reply:
ahh, makes senese. Thanks Roger for digging into that!

Reply:
Will just need to upgrade `llama-index-agent-openai` in a future PR to use the `Thread` wrapper.

Reply:
This PR already updated all the import Thread statements via search and replaces.

Reply:
I should have also bumped the minimum version requirements on the `llama-index-core`. Sorry about that.

Reply:
Let me submit a new PR for the version bump.

Reply:
Thanks. That's my miss too for not noticing.